benchmark/README.md

```diff
@@ -1,45 +1,104 @@
+
+macOS benchmark driver binaries are placed in ``bin`` alongside ``swiftc``.
+Additional platform binaries are placed in the ``benchmark/bin`` build
+directory.
+
+**NOTE:** We assume that the user has installed all Swift project dependencies
+since we will build the rest of the compiler as well.
+
+### Building standalone with SwiftPM
+
+To build the benchmark suite with SwiftPM just:
+
+    $ cd ./swift/benchmark
+    $ swift build
+
+This can be used to generate an Xcode project that can be used to edit the
+benchmarks:
 
-Building with build-script
---------------------------
+    $ cd ./swift/benchmark
+    $ swift package generate-xcodeproj
+    $ open swiftbench.xcodeproj
 
-By default, Swift benchmarks for OS X are compiled during the Swift build
-process. To build Swift benchmarks for additional platforms, pass the following
-flags:
+**NOTE:** By default the Xcode project's scheme for SwiftBench will build the
+benchmark in debug. One must change it to build the benchmark suite in release
+mode.
 
-    $ swift/utils/build-script --ios --watchos --tvos
+**NOTE:** To add a new benchmark, one can not add the benchmark via Xcode itself
+since we generate the Xcode project from the main ``Package.swift`` in
+``./swift/benchmark``. For information on how to do this, please see [Adding New
+Benchmarks](#adding-new-benchmarks).
 
-OS X benchmark driver binaries are placed in `bin` alongside `swiftc`.
-Additional platform binaries are placed in the `benchmark/bin` build directory.
+**NOTE:** This is the most convenient way to get up and running/adding
+benchmarks quickly when used in combination with a recent nightly [developer
+snapshot](https://swift.org/download/#snapshots).  Just use the SwiftPM provided
+by the nightly snapshot and everything should work.
 
-Building Independently
-----------------------
+### Building standalone with ``cmake``.
 
 To build the Swift benchmarks using only an Xcode installation: install an
-Xcode version with Swift support, install cmake 2.8.12, and ensure Xcode is
-selected with xcode-select.
+Xcode version with Swift support, install ``cmake`` 2.8.12, and ensure Xcode is
+selected with ``xcode-select``.
 
-The following build options are available:
+There are two ways to build this. The first is against a specific swift
+snapshot:
+
+    $ cmake -G Ninja \
+        -DSWIFT_EXEC=$PATH_TO_SWIFTC \
+        -DSWIFT_DARWIN_XCRUN_TOOLCHAIN=XcodeDefault \
+        ./swift/benchmark
+
+This will configure the benchmarks to build using the specified ``swiftc``, but
+use the ``clang``/SDK from the XcodeDefault toolchain of the installed cmake.
+
+The second involves specifying by hand all of the paths that ``cmake`` needs rather
+than relying on inference via ``SWIFT_DARWIN_XCRUN_TOOLCHAIN``. Specifically:
+
+    $ cmake -G Ninja \
+        -DSWIFT_EXEC=$SWIFT_BUILD_DIR/bin/swiftc \
+        -DSWIFT_LIBRARY_PATH=$SWIFT_BUILD_DIR/lib/macosx \
+        -DCLANG_EXEC=$LLVM_BUILD_DIR/bin/clang \
```

Comment:
This one didn't work for me. I've set the both variables to point to the corresponding build directory, but the `LLVM_BUILD_DIR` isn't being picked up correctly:
````
CMake Error at cmake/modules/SwiftBenchmarkUtils.cmake:61 (message):
  Unable to find Clang driver
Call Stack (most recent call first):
  cmake/modules/AddSwiftBenchmarkSuite.cmake:27 (runcmd)
  CMakeLists.txt:254 (configure_build)
````
which seem's to try to infer the CLANG_EXEC again from SWIFT_DARWIN_XCRUN_TOOLCHAIN. I think there's bug in AddSwiftBenchmarkSuite.cmake. Line 24 should IMO include `if(NOT CLANG_EXEC)` check similar the the `if(NOT SWIFT_EXEC)` on line 14.